core/src/edu/asu/c3simulator/screens/FundingScreen.java

```diff
@@ -49,29 +50,27 @@ public void clicked(InputEvent event, float x, float y)
 				- (0.01f * stage.getWidth()), (0.01f * stage.getWidth()));
 		
 		// TODO add Corner Advisor
-		// TODO add Home Button
+		HomeButton home = new HomeButton(game);
 		
-		NavigationPanel navigationPanel = createNavigationPanel();
-		
-		stage.addActor(navigationPanel);
+		stage.addActor(home);
 		stage.addActor(continueButton);
 		stage.addActor(ScreenDiscription);
 	}
 	
 	@Override
-	public NavigationPanel createNavigationPanel()
+	public void createNavigationPanel()
```

Comment:
@malyahy2 The problem then, is that the navigationPanel cannot be created in the constructor, because it requires instances of other screens in order to be created, but the other screens cannot be created before the navigation panel. The solution, was the enumerated singleton, AllManagementScreens, in addition to deferred instantiation, via instantiate() methods. Moving the navigation panel creation to the instantiate() method will solve this problem. Alternatively, the navigationPanel could take an AllManagementScreens instance (e.g., AllManagementScreens.INDUSTRY) as an argument, instead of a Screen, and then direct game to screen.getInstance();

The latter is the cleanest solution proposed so far, followed by the former. However, publicizing createNavigationPanel() and using that method to add to the stage directly adds unnecessary bloat, and creates inconsistencies.